samples/sample13/src/drink_store/drink_store.bal

```diff
@@ -101,47 +101,51 @@ service DrinkStoreAPI on drinkStoreEP {
     }
 }
 
-function getCoolDrinkPrice(json payload) returns (json){
+function getCoolDrinkPrice(json payload) returns json {
     json[] items = <json[]>payload;
     float celciusValue = getTempreatureInCelcius();
     foreach json item in items {
         float|error result = <float>item.price;
         if (result is float) {
             float priceVariation = roundFloat(result * (celciusValue / 100), 2);
-            if (celciusValue > 15) {
-                // Increase Cooldrink price on hot days.
-                item.price = roundFloat((result + priceVariation), 2);
-                item.diff = "+" + priceVariation;
-            } else {
-                // Decrese Cooldrink price on cold days.
-                item.price = roundFloat((result - priceVariation), 2);
-                item.diff = "-" + priceVariation;
+            if (item is map<json>) {
+                if (celciusValue > 15) {
+                    // Increase Cooldrink price on hot days.
+                    item["price"] = roundFloat((result + priceVariation), 2);
+                    item["diff"] = string`+ ${priceVariation}`;
+                } else {
+                    item["price"] = roundFloat((result + priceVariation), 2);
+                    item["diff"] = string`- ${priceVariation}`;
+                    // Decrese Cooldrink price on cold days.
```

Comment:
This comment needs to be above `item["price"] = roundFloat((result + priceVariation), 2);` line

Reply:
Fixed